MSAL/src/instance/MSALAuthority.m

```diff
@@ -157,7 +157,9 @@ + (void)resolveEndpointsForAuthority:(NSURL *)unvalidatedAuthority
         tenant = firstPathComponent;
     }
     
-    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority userPrincipalName:userPrincipalName];
+    MSALAuthority *authorityInCache = [MSALAuthority authorityFromCache:updatedAuthority
+                                                          authorityType:authorityType
+                                                      userPrincipalName:userPrincipalName];
     
```

Comment:
Validation portion is in the openIDConfigurationEndpointForAuthority: of the MSAL___AuthorityResolver class. What we always do is discovery portion.